src/lib/pipeline/m2/material/index.js

```diff
@@ -0,0 +1,256 @@
+    // Flag 0x01 (unlit)
+    if (renderFlags & 0x01) {
+      this.uniforms.lightModifier = { type: 'f', value: '0.0' };
+    }
+
+    // Flag 0x02 (unfogged)
+    if (renderFlags & 0x02) {
+      this.uniforms.fogModifier = { type: 'f', value: '0.0' };
+    }
+
+    // Flag 0x04 (no backface culling)
+    if (renderFlags & 0x04) {
+      this.side = THREE.DoubleSide;
+      this.transparent = true;
+    }
+
+    // Flag 0x10 (no z-buffer write)
+    if (renderFlags & 0x10) {
+      this.depthWrite = false;
+    }
+  }
+
+  applyBlendingMode(blendingMode) {
+    if (blendingMode === 1) {
+      this.uniforms.alphaKey = { type: 'f', value: 1.0 };
+    } else {
+      this.uniforms.alphaKey = { type: 'f', value: 0.0 };
+    }
+
+    if (blendingMode >= 1) {
+      this.transparent = true;
+      this.blending = THREE.CustomBlending;
+    }
+
+    switch (blendingMode) {
+      case 0:
+        this.blending = THREE.NoBlending;
+        this.blendSrc = THREE.OneFactor;
+        this.blendDst = THREE.ZeroFactor;
+        break;
+
+      case 1:
+        this.alphaTest = 0.5;
+        this.side = THREE.DoubleSide;
+
+        this.blendSrc = THREE.OneFactor;
+        this.blendDst = THREE.ZeroFactor;
+        this.blendSrcAlpha = THREE.OneFactor;
+        this.blendDstAlpha = THREE.ZeroFactor;
+        break;
+
+      case 2:
+        this.blendSrc = THREE.SrcAlphaFactor;
+        this.blendDst = THREE.OneMinusSrcAlphaFactor;
+        this.blendSrcAlpha = THREE.SrcAlphaFactor;
+        this.blendDstAlpha = THREE.OneMinusSrcAlphaFactor;
+        break;
+
+      case 3:
+        this.blendSrc = THREE.SrcColorFactor;
+        this.blendDst = THREE.DstColorFactor;
+        this.blendSrcAlpha = THREE.SrcAlphaFactor;
+        this.blendDstAlpha = THREE.DstAlphaFactor;
+        break;
+
+      case 4:
+        this.blendSrc = THREE.SrcAlphaFactor;
+        this.blendDst = THREE.OneFactor;
+        this.blendSrcAlpha = THREE.SrcAlphaFactor;
+        this.blendDstAlpha = THREE.OneFactor;
+        break;
+
+      case 5:
+        this.blendSrc = THREE.DstColorFactor;
+        this.blendDst = THREE.ZeroFactor;
+        this.blendSrcAlpha = THREE.DstAlphaFactor;
+        this.blendDstAlpha = THREE.ZeroFactor;
+        break;
+
+      case 6:
+        this.blendSrc = THREE.DstColorFactor;
+        this.blendDst = THREE.SrcColorFactor;
+        this.blendSrcAlpha = THREE.DstAlphaFactor;
+        this.blendDstAlpha = THREE.SrcAlphaFactor;
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  loadTextures() {
+    const textureDefs = this.textureDefs;
+
+    const textures = [];
+
+    textureDefs.forEach((textureDef) => {
+      textures.push(this.loadTexture(textureDef));
+    });
+
+    // Update shader uniforms to reflect loaded textures.
+    this.uniforms.textures = { type: 'tv', value: textures };
+    this.uniforms.textureCount = { type: 'i', value: textures.length };
+  }
+
+  loadTexture(textureDef) {
+    let loaded = null;
```

Comment:
Agreed! In general, I'd like to revisit M2s when we do blueprints. I'm hoping some of the less straightforward code paths will be much more straightforward when blueprints come in to being.